src/main/resources/script.sh

```diff
@@ -0,0 +1,1353 @@
+done
+
+# Stop log for application environment
+stopLog application_environment
+
+
+function download_udocker {
+  #installation of udocker
+  info "cloning udocker ${UDOCKER_TAG} "
+  git clone --depth=1 --branch ${UDOCKER_TAG} https://github.com/indigo-dc/udocker.git
+  (cd udocker/udocker; ln -s maincmd.py udocker)
+  export PATH=`pwd`/udocker/udocker:$PATH
+  
+  #creating a temporary directory for udocker containers
+  mkdir -p containers
+  export UDOCKER_CONTAINERS=$PWD/containers
+  
+  #find pre-deployed containers on CVMFS, and create a symlink to the udocker containers directory
+  ## use a global velocity escape to avoid velocity escaping issue
+  for d in ${CONTAINERS_CVMFS_PATH}/*/ ;
+     do mkdir containers/$(basename "${d%/}") && ln -s "${d%/}"/* containers/$(basename "${d%/}")/
+  done
+  cat >docker <<'EOF'
+        #!/bin/bash
+        MYARGS=$*
+        echo "executing ./udocker/udocker/udocker $MYARGS"
+        ./udocker/udocker/udocker $MYARGS
+EOF
+  chmod a+x docker
+  export PATH=$PWD:$PATH
+}
+
+if ! command -v docker
+then
+    download_udocker
+fi
+####################################################################################################
+####################################################################################################
+function checkBosh {
+  local BOSH_CVMFS_PATH=$1
+  #by default, use CVMFS bosh
+  ${BOSH_CVMFS_PATH}/bosh create foo.sh
+  if [ $? != 0 ]
+  then
+    info "CVMFS bosh in ${BOSH_CVMFS_PATH} not working, checking for a local version"
+    bosh create foo.sh
+    if [ $? != 0 ]
+    then
+        info "bosh is not found in PATH or it is does not work fine, searching for another local version"
+        local HOMEBOSH=`find $HOME -name bosh`
+        if [ -z "$HOMEBOSH" ]
+        then
+            info "bosh not found, trying to install it"
+            pip install --trusted-host pypi.org --trusted-host pypi.python.org --trusted-host files.pythonhosted.org boutiques --prefix $PWD 
+            if [ $? != 0 ]
+            then
+                error "pip install boutiques failed"
+                exit 1
+            else
+                export BOSHEXEC="$PWD/bin/bosh"
+            fi
+        else
+            info "local bosh found in $HOMEBOSH"
+            export BOSHEXEC=$HOMEBOSH
+        fi
+    else # bosh is found in PATH and works fine
+        info "local bosh found in $PATH"
+        export BOSHEXEC="bosh"
+    fi
+  else # if bosh CVMFS works fine
+    export BOSHEXEC="${BOSH_CVMFS_PATH}/bosh"
+  fi
+}
+checkBosh $BOSH_CVMFS_PATH
+####################################################################################################
+function boutiques_exec {
+echo "import sys; sys.setdefaultencoding(\"UTF8\")" > sitecustomize.py
+COMMAND_LINE="PYTHONPATH=".:$PYTHONPATH" $BOSHEXEC exec launch -x $jsonFileName ../inv/$invocationJson -v $PWD/../cache:$PWD/../cache"
+# Execute the command and store the output in a temporary file
+eval "$COMMAND_LINE" > temp_output.txt
```

Comment:
I used it to read the name of the provenance file, but since we would now use the copy_provenance function, the whole code block regarding reading the output of COMMAND_LINE and fetching the provenance file is redundant.